src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java

```diff
@@ -113,30 +118,43 @@ public void replayMilestones(Snapshot snapshot, int targetMilestoneIndex) throws
                 if (currentMilestone != null) {
                     StateDiffViewModel stateDiffViewModel = StateDiffViewModel.load(tangle, currentMilestone.getHash());
                     if(!stateDiffViewModel.isEmpty()) {
-                        snapshot.applyStateDiff(new SnapshotStateDiffImpl(stateDiffViewModel.getDiff()));
+                        stateDiffViewModel.getDiff().forEach((address, balance) -> {
+                            if (balanceChanges.computeIfPresent(address, (hash, value) -> balance + value) == null) {
```

Comment:
I know this is the way state updates are written in IRI, but it's not easy to reason about.
could we consider:
1.
```suggestion
balanceChanges.computeIfPresent(address, (hash, value) -> balance + value);
balanceChanges.putIfAbsent(address, balance);
```
2.
```suggestion
balanceChanges.compute(address, (hash, value) -> (value == null) ? balace : balance + value)
```
3.
```suggestion
balanceChanges.put(address, balance + balanceChanges.getOrDefault(value, 0L))
```

Reply:
rewrote it as proposed